qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1382,12 +1380,9 @@ def _on_load_finished(self, ok):
                 emit_before_load_started=False))
             self._reload_url = None
 
-        if not qtutils.version_check('5.10', compiled=False):
-            # We can't do this when we have the loadFinished workaround as that
-            # sometimes clears icons without loading a new page.
-            # In general, this is handled by Qt, but when loading takes long,
-            # the old icon is still displayed.
-            self.icon_changed.emit(QIcon())
+        # In general, this is handled by Qt, but when loading takes long,
+        # the old icon is still displayed.
+        self.icon_changed.emit(QIcon())
```

Comment:
After a bit of testing, I'm finding a couple pages show an empty QIcon after loading instead of the site one. I don't think the ordering of the signals is stable still, even on initial page load.

Reply:
Here's a stack of a load where the icon is cleared erroneously. Is it possible to just clear this on loadStarted instead maybe? Or as a worst-case, restore the old workaround (annoying but it at least ensured the favicon is present normally).

```
Setting tab icon: <PyQt5.QtGui.QIcon object at 0x7fc916fa99d8>, null: True
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jay/Code/qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "/home/jay/Code/qutebrowser/qutebrowser/qutebrowser.py", line 196, in main
    return app.run(args)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 144, in run
    ret = qt_mainloop()
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 154, in qt_mainloop
    return q_app.exec_()
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 668, in on_icon_changed
    self.widget.setTabIcon(idx, icon)
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabwidget.py", line 360, in setTabIcon
    traceback.print_stack()
Setting tab icon: <PyQt5.QtGui.QIcon object at 0x7fc916fa99d8>, null: False
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jay/Code/qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "/home/jay/Code/qutebrowser/qutebrowser/qutebrowser.py", line 196, in main
    return app.run(args)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 144, in run
    ret = qt_mainloop()
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 154, in qt_mainloop
    return q_app.exec_()
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 668, in on_icon_changed
    self.widget.setTabIcon(idx, icon)
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabwidget.py", line 360, in setTabIcon
    traceback.print_stack()
webenginetab: on_load_finished
Setting tab icon: <PyQt5.QtGui.QIcon object at 0x7fc916fa9c18>, null: True
  File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jay/Code/qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "/home/jay/Code/qutebrowser/qutebrowser/qutebrowser.py", line 196, in main
    return app.run(args)
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 144, in run
    ret = qt_mainloop()
  File "/home/jay/Code/qutebrowser/qutebrowser/app.py", line 154, in qt_mainloop
    return q_app.exec_()
  File "/home/jay/Code/qutebrowser/qutebrowser/browser/webengine/webenginetab.py", line 1396, in _on_load_finished
    self.icon_changed.emit(QIcon())
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabbedbrowser.py", line 668, in on_icon_changed
    self.widget.setTabIcon(idx, icon)
  File "/home/jay/Code/qutebrowser/qutebrowser/mainwindow/tabwidget.py", line 360, in setTabIcon
    traceback.print_stack()
```

I wish we can get proper tracebacks with signals, but it seems like iconChanged is coming before loadFinished from webengine.

Reply:
Meh, this really introduces more issues than it solves to be honest - given that nobody on Qt >= 5.10 (where this is inactive) complained in the past 1.5 years, I just removed it entirely in c369257acc5aeda968f59a6f88b78de92f4db22b.